exp/api/v1alpha4/azuremanagedcontrolplane_webhook.go

```diff
@@ -306,3 +312,71 @@ func (r *AzureManagedControlPlane) validateSSHKey() error {
 
 	return nil
 }
+
+// validateAPIServerAccessProfile validates an APIServerAccessProfile.
+func (r *AzureManagedControlPlane) validateAPIServerAccessProfile() error {
+	if r.Spec.APIServerAccessProfile != nil {
+		var allErrs field.ErrorList
+		for _, ipRange := range r.Spec.APIServerAccessProfile.AuthorizedIPRanges {
+			if _, _, err := net.ParseCIDR(ipRange); err != nil {
+				allErrs = append(allErrs, field.Invalid(field.NewPath("Spec", "APIServerAccessProfile", "AuthorizedIPRanges"), ipRange, "invalid CIDR format"))
+			}
+		}
+		if len(allErrs) > 0 {
+			agg := kerrors.NewAggregate(allErrs.ToAggregate().Errors())
+			azuremanagedcontrolplanelog.Info("Invalid apiServerAccessProfile: %s", agg.Error())
+			return agg
+		}
+	}
+	return nil
+}
+
+// validateAPIServerAccessProfileUpdate validates update to APIServerAccessProfile.
+func (r *AzureManagedControlPlane) validateAPIServerAccessProfileUpdate(old *AzureManagedControlPlane) field.ErrorList {
```

Comment:
There are quite a few field combinations here, it would be good to have unit tests for this.

Also, maybe using `DeepEqual` would make this shorter, like here for example https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/v0.5.2/api/v1alpha4/azurecluster_webhook.go#L109. @alexeldeib @CecileRobertMichon is using `DeepEqual` for comparison recommended/acceptable, or is handpicked field comparison like here more preferable?

Reply:
Sure, I'll add unit tests for this.

Unfortunately I cannot use `DeepEqual` here since one of the field (`AuthorizedIPRanges`) is mutable. One option is to move `AuthorizedIPRanges` out of `APIServerAccessProfile` so we can use `DeepEqual` here. I don't have a strong preference here. Would love to know what do people think.

Reply:
You could normalize values before comparison, i.e. set mutable fields to empty/`nil` values in both operands (you would have to copy them first and compare normalized copies to avoid operand mutation).

No strong opinion for using `DeepEqual`, just a suggestion.

Reply:
Thanks for the suggestion! I like that, it makes the code much simpler.

The downside is that it will prevent certain updates that is validate (e.g. change `EnablePrivateCluster` from `nil` to a pointer to `false`).

I've updated the code to use `DeepEqual`, but I'm open to change it back if people have strong opinions.

Reply:
Just a suggestion, one way to avoid DeepEqual while comparing is use cmp.Diff from github.com/google/go-cmp/cmp.

Reply:
cmp uses reflection quite a bit, it just wraps some of the tricky bits of comparisons nicely.

Reply:
ohh okay